packages/blade/src/components/Input/SearchInput/_decisions/decisions.md

```diff
@@ -0,0 +1,184 @@
+  ActionListItem,
+  Box,
+} from '@razorpay/blade/components';
+
+const App = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const menuItems = [
+    'Settings',
+    'Profile',
+    'Transactions',
+    'Help',
+    'Refunds',
+    'Settlements',
+    'Payouts',
+  ];
+  const popularItems = ['Transactions', 'Settlements'];
+
+  return (
+    <Dropdown>
+      <SearchInput
+        label="Search"
+        onChange={({ value }) => setSearchTerm(value)}
+        placeholder="Search here"
+      />
+      <DropdownOverlay>
+        <ActionList>
+          {searchTerm.length === 0
+            ? popularItems.map((item, index) => (
+                <ActionListItem key={index} title={item} value={item} />
+              ))
+            : menuItems
+                .filter((item) => item.includes(searchTerm))
+                .map((item, index) => <ActionListItem key={index} title={item} value={item} />)}
+        </ActionList>
+      </DropdownOverlay>
+    </Dropdown>
+  );
+};
+```
+
+### With Table
+
+```jsx
+import React from 'react';
+import { Box, SearchInput } from '@razorpay/blade/components';
+
+const App = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  return (
+    <Table
+      data={data}
+      toolbar={
+        <TableToolbar>
+          <TableToolbarActions>
+            <SearchInput
+              label="Search bank transactions"
+              onChange={({ value }) => setSearchTerm(value)}
+              placeholder="SBI, HDFC, ICICI, etc."
+            />
+          </TableToolbarActions>
+        </TableToolbar>
+      }
+    >
+      {(tableData) => (
+        <>
+          <TableHeader>
+            <TableHeaderRow>
+              <TableHeaderCell>Bank Name</TableHeaderCell>
+              <TableHeaderCell>Amount</TableHeaderCell>
+              <TableHeaderCell>Date</TableHeaderCell>
+              <TableHeaderCell>Method</TableHeaderCell>
+            </TableHeaderRow>
+          </TableHeader>
+          <TableBody>
+            {tableData
+              // Filter item based on the search input value
+              .filter((tableItem) => tableItem.bankName.includes(searchTerm))
+              .map((tableItem, index) => (
+                <TableRow key={index} item={tableItem}>
+                  <TableCell>
+                    <Code size="medium">{tableItem.paymentId}</Code>
+                  </TableCell>
+                  <TableCell>
+                    <Amount value={tableItem.amount} />
+                  </TableCell>
+                  <TableCell>
+                    {tableItem.date?.toLocaleDateString('en-IN', {
+                      year: 'numeric',
+                      month: '2-digit',
+                      day: '2-digit',
+                    })}
+                  </TableCell>
+                  <TableCell>{tableItem.status}</TableCell>
+                </TableRow>
+              ))}
+          </TableBody>
+        </>
+      )}
+    </Table>
+  );
+};
+```
+
```

Comment:
Add references to how other design systems handle their SearchInput if there are any?